### ISSUE TITLE
Please help me, I'm new to this. Provide me with the code for Schlumberger VES inversion with field data

### ISSUE THREAD

**Issue by ferryatm** [gimli-org/gimli#673]
[Datasheet TA.xlsx](https://github.com/gimli-org/gimli/files/14606947/Datasheet.TA.xlsx)
### Discussed in https://github.com/gimli-org/gimli/discussions/672

<div type='discussions-op-text'>

<sup>Originally posted by **ferryatm** March 15, 2024</sup>
Please help me, I'm new to this. Provide me with the code for Schlumberger VES inversion with field data.

| MN/2 (m) | AB/2 (m) | Rho Apparent (Ohm.m) |
|----------|----------|-------------------------|
|   0.5    |   1.5    |         5.635211        |
|   0.5    |   2.0    |         4.612248        |
|   0.5    |   2.5    |         4.513198        |
|   0.5    |   3.0    |         4.17725         |
|   0.5    |   4.0    |         5.30145         |
|   0.5    |   5.0    |         4.7350773       |
|   0.5    |   6.0    |         4.3606916       |
|   1.0    |   6.0    |         4.3606916       |
|   1.0    |   7.0    |         4.614735575     |
|   1.0    |   8.0    |         5.28781021      |
|   1.0    |   9.0    |         5.463961275     |
|   1.0    |   10.0   |         5.334367416     |
|   1.0    |   12.0   |         5.871748295     |
|   1.0    |   15.0   |         6.864989891     |
|   5.0    |   15.0   |         6.864989891     |
|   5.0    |   20.0   |         8.31374329      |
|   5.0    |   25.0   |         10.45239054     |
|   5.0    |   30.0   |         11.65316963     |
|   10.0   |   30.0   |         11.65316963     |
|   10.0   |   40.0   |         13.94573528     |
|   10.0   |   50.0   |         16.34242351     |
|   10.0   |   60.0   |         17.40842728     |
|   20.0   |   60.0   |         17.40842728     |
|   20.0   |   70.0   |         20.31610704     |
|   20.0   |   80.0   |         19.13295652     |
|   20.0   |   90.0   |         20.17732947     |
|   20.0   |   100.0  |         20.20464984     |
</div>

**Comment by ferryatm** [comment]
Please help me just this once

**Comment by halbmy** [comment]
Here's some code. Most things are, however, about reading and filtering data using numpy and pandas.
```
import pygimli as pg
from pygimli.physics import ves
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
xls = pd.read_excel("Datasheet.TA.xlsx", skiprows=9)
print(xls)
xls = xls.dropna(subset="Rho (Ohm.m)")
ab2 = xls["AB/2 (m)"].array
mn2 = xls["MN/2 (m)"].array
rhoa = xls["Rho (Ohm.m)"].array
mgr = ves.VESManager()
mgr.fop.setDataSpace(ab2=ab2, mn2=mn2)
mgr.invert(rhoa, err=np.ones_like(rhoa)*0.05, verbose=True)
fig, ax = plt.subplots()
ax.loglog(rhoa, ab2, "x", label="data")
ax.loglog(mgr.inv.response, ab2, "+-", label="response")
ax.grid()
ax.legend()
ax.invert_yaxis()
```
![image](https://github.com/gimli-org/gimli/assets/7136401/5cd30cda-97b1-4265-8ba6-67d1bdc1461c)

```
fig, ax = plt.subplots()
pg.viewer.mpl.drawModel1D(ax, model=model)
```
![image](https://github.com/gimli-org/gimli/assets/7136401/b7b4196c-c313-42fd-b842-0a326b35ac92)

**Comment by ferryatm** [comment]
Thank you very much. May I ask for one more assistance? Can this VES data be used for 2D modeling? If so, please kindly provide the code. I truly apologize if this causes any inconvenience.

**Comment by ferryatm** [comment]
![image](https://github.com/gimli-org/gimli/assets/102302050/ea75e0c3-0332-473b-b062-34ef13c2c40e)

please I have an error regarding this, same with np.bool, np.long and others, it's like an outdated version of numpy, is there any solution?

**Comment by ferryatm** [comment]
This is related to the code you provided

**Comment by halbmy** [comment]
This seems to be an outdated numpy version, which one do you have?

**Comment by halbmy** [comment]
> Can this VES data be used for 2D modeling?

Generally, of course. However, it makes only sense for a couple of (overlapping) soundings. You would have to collect the data in a `DataContainerERT`. In case of separated soundings, an LCI inversion is more appropriate.

**Comment by ferryatm** [comment]
Numpy version 1.24.4

**Comment by halbmy** [comment]
This is a quite modern version. It seems like you are using an outdated pyGIMLi version (which one?) that is no more compatible with modern numpy. The problem was solved with v1.3.1. Try update pyGIMLi (or downgrade numpy).

**Comment by ferryatm** [comment]
Thank you in advance, I might give it a try first, but please don't close it temporarily because I really need guidance and assistance as a newcomer

________________________________
Dari: Thomas Günther ***@***.***>
Dikirim: Jumat, 15 Maret 2024 15.54
Kepada: gimli-org/gimli ***@***.***>
Cc: FERRY ATH THAARIQ MUDHOFIR ***@***.***>; Author ***@***.***>
Subjek: Re: [gimli-org/gimli] Please help me, I'm new to this. Provide me with the code for Schlumberger VES inversion with field data (Issue #673)


This is a quite modern version. It seems like you are using an outdated pyGIMLi version (which one?) that is no more compatible with modern numpy. The problem was solved with v1.3.1. Try update pyGIMLi (or downgrade numpy).

**Comment by halbmy** [comment]
As the original issue had been solved and the last problems are not related to the title and are probably due to an outdated pyGIMLi version, I am closing the issue. In case of any problem with a modern version, please raise a new issue.
